tests/hypothesis/lookup_test.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+import numpy as np
+from hypothesis import given
+from hypothesis import strategies as st
+
+import scipp as sc
+
+float_args = dict(min_value=-1e300,
+                  max_value=1e300,
+                  allow_nan=False,
+                  allow_infinity=False,
+                  allow_subnormal=False)
+
+
+@given(st.floats(**float_args), st.floats(**float_args))
+def test_lookup_linspace_bounds(a, b):
+    x = sc.array(dims=['row'], values=[a, b])
+    upper = x.max()
+    upper.value = np.nextafter(upper.value, np.inf)
+    x_edges = sc.linspace('x', x.min(), upper, num=3)
+    hist = sc.DataArray(sc.array(dims=['x'], values=[1.1, 1.1]), coords={'x': x_edges})
+    func = sc.lookup(hist, 'x', fill_value=sc.scalar(1234.0))
+
+    result = func(x)
+
+    assert sc.identical(result, sc.full_like(x, value=1.1))
+
+
+@given(st.floats(**float_args), st.floats(**float_args))
+def test_lookup_scale_linspace_bounds(a, b):
+    x = sc.array(dims=['row'], values=[a, b])
+    upper = x.max()
+    upper.value = np.nextafter(upper.value, np.inf)
+    x_edges = sc.linspace('x', x.min(), upper, num=3)
+    hist = sc.DataArray(sc.array(dims=['x'], values=[1.1, 1.1]), coords={'x': x_edges})
+    func = sc.lookup(hist, 'x', fill_value=sc.scalar(1234.0))
+
+    da = x.bin(x=x_edges[::2].copy())
+    da.bins *= func
+
+    assert da.sum().value == 2.2
```

Comment:
I'm not sure what is being tested in these two cases? It looks like all bins are in bounds and the fixed out-of-bounds case is not being tested?

Reply:
The bug that is fixed was that the old implementation detected values out of bounds even if they were not.

Are you asking about the reverse, i.e., event that are out of bounds that would be interpreted as in bounds?

Reply:
I guess it works either way: Values getting larger or smaller. I suppose since we have an upper and lower bin bound we are kind of covering both cases here?

Reply:
My point is that due to the bug in the previous implementation a value may have shifted up or down. Since we have values at both bounds, this test should detect both cases?

Reply:
ok